hyppo/kgof/_utils.py

```diff
@@ -0,0 +1,310 @@
+"""A module containing convenient methods"""
+from __future__ import print_function
+from __future__ import division
+from __future__ import unicode_literals
+from __future__ import absolute_import
+
+from builtins import zip
+from builtins import int
+from builtins import range
+from future import standard_library
+standard_library.install_aliases()
+from past.utils import old_div
+from builtins import object
+
+import autograd.numpy as np
+import time
+
+class ContextTimer(object):
+    """
+    A class used to time an execution of a code snippet. 
+    Use it with with .... as ...
+    For example, 
+        with ContextTimer() as t:
+            # do something 
+        time_spent = t.secs
+    From https://www.huyng.com/posts/python-performance-analysis
+    """
+
+    def __init__(self, verbose=False):
+        self.verbose = verbose
+
+    def __enter__(self):
+        self.start = time.time()
+        return self
+
+    def __exit__(self, *args):
+        self.end = time.time()
+        self.secs = self.end - self.start 
+        if self.verbose:
+            print('elapsed time: %f ms' % (self.secs*1000))
+
+# end class ContextTimer
+
+class NumpySeedContext(object):
```

Comment:
Can I use `numpy.random.set_state() ` instead? Works more fluidly with what I've written.

Reply:
Biggest reason I chose to keep this method is that I need something that sets the seed back to the current random state at the end of the block I wanted to execute using the given seed.

Reply:
`random_state` is used by other functions in hyppo. That's why I mentioned that, just for consistency

Reply:
That makes sense, given I need to set the seed back to what it originally was following the block, is there any way this can stay? Or should I try to rewrite?

Reply:
just rewrite using a `rng` object. See `numpy.random` to see how that works

Reply:
Ripped and replaced